homeassistant/components/thingspeak.py

```diff
@@ -43,7 +43,7 @@ def setup(hass, config):
 
     try:
         channel = thingspeak.Channel(
-            channel_id, api_key=api_key, timeout=TIMEOUT)
+            channel_id, write_key=api_key, timeout=TIMEOUT)
```

Comment:
You don't need to change the attribute name. You need to use correct key (as value